CMakeLists.txt

```diff
@@ -144,6 +145,18 @@ add_link_options(
     -fno-sanitize-recover=all
     -fno-omit-frame-pointer
 )
+elseif(CONFIG_TSAN)
+message(STATUS "Building with TSan")
+add_compile_options(
+    -fsanitize=thread
+    -fno-sanitize-recover=all
```

Comment:
Unrelated to this PR, but we do we set this option to recover while at the same time setting the one to halt on any error?

Reply:
It's because of #22, reported by you :)

Reply:
Ah sorry I misread, we set the NO recover option 😅